client/my-sites/patterns/controller.tsx

```diff
@@ -0,0 +1,63 @@
+import type { Context as PageJSContext } from '@automattic/calypso-router';
+import wpcom from 'calypso/lib/wp';
+import { getCurrentUserLocale, isUserLoggedIn } from 'calypso/state/current-user/selectors';
+import { getPatternSourceSiteID } from 'calypso/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/constants';
+import type { Pattern } from 'calypso/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/types';
+import LoggedInComponent from './logged-in';
+import LoggedOutComponent from './logged-out';
+
+type Next = ( error?: Error ) => void;
+
+export const fetchPatterns = ( context: PageJSContext, next: Next ) => {
+	if ( context.cachedMarkup ) {
+		return next();
+	}
+
+	const currentUserLocale = getCurrentUserLocale( context.store.getState() ) || context.lang;
+
+	const locale = currentUserLocale || 'en';
+
+	context.queryClient
+		.fetchQuery( {
+			queryKey: [ locale, 'patterns' ],
+			queryFn: () => {
+				return wpcom.req.get( `/ptk/patterns/${ locale }`, {
+					categories: 'intro',
+					per_page: '5',
+					post_type: 'wp_block',
+					site: getPatternSourceSiteID(),
+				} );
+			},
+			staleTime: Infinity,
+		} )
```

Comment:
I initially wondered why we fetch this data with `queryClient` on the back-end. Turns out, per [these docs](https://github.com/Automattic/wp-calypso/blob/trunk/docs/isomorphic-routing.md#network-requests), that there is good reason for this. Namely, caching!

If we use `useDotcomPatterns` on the front-end, `react-query` will already see an entry in its cache and won't run another network request.